services/weblate/weblate-project-translated-percentage.service.js

```diff
@@ -0,0 +1,72 @@
+'use strict'
+
+const Joi = require('joi')
+const { BaseJsonService } = require('..')
+const { optionalUrl } = require('../validators')
+const { colorScale } = require('../color-formatters')
+
+const schema = Joi.object({
+  translated_percent: Joi.number().required(),
+}).required()
+
+const queryParamSchema = Joi.object({
+  server: optionalUrl.required(),
+}).required()
+
+const documentation = `
+  <p>
+    This badge displays the percentage of strings translated on a project on a
+    Weblate instance.
+  </p>
+`
+
+module.exports = class WeblateProjectTranslatedPercentage extends (
+  BaseJsonService
+) {
+  static category = 'other'
+  static route = { base: 'weblate', pattern: ':project', queryParamSchema }
+
+  static examples = [
+    {
+      title: 'Weblate project translated',
+      namedParams: { project: 'godot-engine' },
+      queryParams: { server: 'https://hosted.weblate.org' },
+      staticPreview: this.render({ translatedPercent: 20.5 }),
+      documentation,
+      keywords: ['i18n', 'translation', 'internationalization'],
+    },
+  ]
+
+  static defaultBadgeData = { label: 'translated' }
+
+  /**
+   * Takes a percentage and maps it to a message and color.
+   *
+   * The colors are determined based on how Weblate does it internally.
+   * {@link https://github.com/WeblateOrg/weblate/blob/main/weblate/trans/widgets.py Weblate on GitHub}
+   *
+   * @param {*} translatedPercent The percentage of translations translated.
+   * @returns {object} Format for the badge.
+   */
+  static render({ translatedPercent }) {
+    const color = colorScale([75, 90])(translatedPercent)
+    return { message: `${translatedPercent}%`, color }
```

Comment:
We do round all our percentages to the nearest whole number via `.toFixed(0)` and should do this here as well for consistency